plugins/base/src/main/kotlin/renderers/PackageListService.kt

```diff
@@ -2,44 +2,43 @@ package org.jetbrains.dokka.base.renderers
 
 import org.jetbrains.dokka.base.DokkaBase
 import org.jetbrains.dokka.links.DRI
-import org.jetbrains.dokka.links.parent
 import org.jetbrains.dokka.pages.*
 import org.jetbrains.dokka.plugability.DokkaContext
 import org.jetbrains.dokka.plugability.plugin
 import org.jetbrains.dokka.plugability.querySingle
 import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
-class PackageListService(val context: DokkaContext) {
+class PackageListService(val context: DokkaContext, val rootPage: RootPageNode) {
 
-    fun formatPackageList(module: RootPageNode, format: String, linkExtension: String): String {
+    fun createPackageList(module: ModulePage, format: String, linkExtension: String): String {
 
         val packages = mutableSetOf<String>()
         val nonStandardLocations = mutableMapOf<String, String>()
 
         val locationProvider =
-            context.plugin<DokkaBase>().querySingle { locationProviderFactory }.getLocationProvider(module)
-
-        fun visit(node: PageNode, parentDris: Set<DRI>) {
+            context.plugin<DokkaBase>().querySingle { locationProviderFactory }.getLocationProvider(rootPage)
 
+        fun visit(node: PageNode) {
             if (node is PackagePage) {
                 node.name
                     .takeUnless { name -> name.startsWith("[") && name.endsWith("]") } // Do not include the package name for declarations without one
                     ?.let { packages.add(it) }
             }
 
             val contentPage = node.safeAs<ContentPage>()
-            contentPage?.dri?.forEach {
-                if (parentDris.isNotEmpty() && it.parent !in parentDris) {
-                    locationProvider.resolve(node)
-                        ?.let { nodeLocation -> nonStandardLocations[it.toString()] = nodeLocation }
-                        ?: context.logger.error("Cannot resolve path for ${node.name}!")
+            contentPage?.dri?.forEach { dri ->
+                val nodeLocation = locationProvider.resolve(node, context = module, skipExtension = true)
+                    ?: run { context.logger.error("Cannot resolve path for ${node.name}!"); null }
+
+                if (locationProvider.guessLocation(dri) != nodeLocation && dri != DRI.topLevel) {
```

Comment:
Please swap the conditions here, `&&` is short-circuit  and there is no need to guess location if dri is top-level

Reply:
Fixed